### PR TITLE
#954 avoid orphaned files when using admin > advanced > replace file

### PR DESCRIPTION
**Pull request by SachaMPS**
This should be a workaround for:
https://github.com/divio/django-filer/issues/954

### CODE REVIEW

**Comment by benzkji** [comment]
not a real workaround, looks ok?! slightly better exception handling could be done, but then that's it - I once read one should not just `except:`, but always provide what you expect to except...

+1 for merging this and wasting less server space ;-)

File: filer/models/filemodels.py [hunk 1]
```diff
@@ -191,6 +191,13 @@ def save(self, *args, **kwargs):
         if self._old_is_public != self.is_public and self.pk:
             self._move_file()
             self._old_is_public = self.is_public
+        # Delete old file(s) when updating the file via: admin > advanced > replace file
+        try:
+            this = File.objects.get(id=self.id)
+            if this.file != self.file:
+                this.file.delete(save=False)
+        except:
```
Comment (yakky) [inline filer/models/filemodels.py]: It's better to explicitly handle the expected exceptions

**Comment by yakky** [comment]
@SachaMPS Thanks for contributing this,, could you rebase on top of current develop?
Sorry it took so long to handle this

**Comment by benzkji** [comment]
how about a merge...1.4 is out :-)

**Comment by jrief** [comment]
@benzkji I fully agree with @yakky on https://github.com/divio/django-filer/pull/958#discussion_r178407709

**Comment by benzkji** [comment]
sure!

**Comment by benzkji** [comment]
is this issue resolved? anyone?

**Comment by merwok** [comment]
No, this PR was not merged (author did not reply to review) and the ticket is still open.

**Comment by benzkji** [comment]
someone would need to handle the exception, and off we go, right?

**Comment by marksweb** [comment]
What are the possible exceptions here. Obviously there's the `DoesNotExist` from the `get()`, but is there something else?

**Comment by benzkji** [comment]
if `this.file` is none (not the usual case, but could be...), this.file.delete() would throw a "None has no method 'delete()'". Rather edge case. Dont see others...

**Comment by benzkji** [comment]
This is important, isn't it?

**Comment by benzkji** [comment]
@jrief or was this alreaedy merge before? It's an easy picking...

**Pull request closed without merging**